.github/CONTRIBUTING.md

```diff
@@ -1,121 +1,120 @@
+The rendered documentation is placed into `docs/build/html`. `index.html` is an entry point.
 
-**Note:** this will mostly result in a lot of udpated files under `tests/cassettes/` don't forget to add them in your PR.
+## Run tests (online)
 
-Add these new files a dedicated commit, in order to make the review process easier please.
+gspread uses [vcrpy](https://github.com/kevin1024/vcrpy) to record and replay HTTP interactions with Sheets API.
 
-The following command will replay existing requests and record new requests:
-```
-GS_RECORD_MODE=new_episodes GS_CREDS_FILENAME=<YOUR_CREDS.json> tox -e py
-```
+### `GS_CREDS_FILENAME` environment variable
 
-Then run the tests in offline mode to make sure you have recorded everything.
+You must provide service account credentials using the `GS_CREDS_FILENAME` environment variable in order to make HTTP requests to the Sheets API.
 
-```
-tox -e py
-```
+[Obtain OAuth2 credentials from Google Developers Console](http://gspread.readthedocs.org/en/latest/oauth2.html).
 
-**Note::** In some cases if the test suite can't record new episodes or it can't
-replay them offline, you can run a complete update of the cassettes using the following command:
+### `GS_RECORD_MODE` environment variable
 
-```
-GS_RECORD_MODE=all GS_CREDS_FILENAME=<YOUR_CREDS.json> tox -e py
-```
+You can control vcrpy's [Record Mode](https://vcrpy.readthedocs.io/en/latest/usage.html#record-modes) using `GS_RECORD_MODE` environment variable. It can be:
 
-3. Format your code:
+- `all` - record all HTTP requests, overwriting existing ones
+- `new_episodes` - record new HTTP requests and replay existing ones
+- `none` - replay existing HTTP requests only
 
-Use the following command to format your code. Doing so will ensure
-all code respects the same format.
+In the following cases, you must record new HTTP requests:
 
-```
-tox -e format
-```
+- a new test is added
+- an existing test is updated and does a new HTTP request
+- gspread is updated and does a new HTTP request
 
-Then run the linter to validate change, if any.
+### Run test, capturing *all* HTTP requests
 
-**Note:** the CI runs that command, if it fail you won't be able to merge
-your changes in GSpread.
+In some cases if the test suite can't record new episodes, or it can't replay them offline, you can run a complete update of the cassettes.
 
-```
-tox -e lint
+```bash
+GS_CREDS_FILENAME=<./YOUR_CREDS.json> GS_RECORD_MODE=all tox -e py
 ```
 
-## Render Documentation
+### Run test, capturing *only new* HTTP requests
 
-The documentation uses [reStructuredText](http://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html#rst-index) markup and is rendered by [Sphinx](http://www.sphinx-doc.org/).
+To record new HTTP requests:
 
-To build the documentation locally, use the following command:
+1. Remove the file holding the recorded HTTP requests of the test(s).
+  (e.g.: for the file `tests/cell_test.py` delete `tests/cassettes/CellTest.json`)
```

Comment:
we create a single cassette file for a single test, so in the `cell_test.py` file we have multiple tests, so it produces multiple cassettes.

Here in order to guide the new user perfectly we should be more prices and mention:
- for the file `tests/cell_test.py`
- for the test `test_a1_value`
- remove the file: `tests/cassettes/CellTest.test_a1_value.json`